libs/stringCore/include/stringCore/string_adapter.h

```diff
@@ -0,0 +1,58 @@
+/**
+
+  Alternate char_traits that remaps wchar_t to wc16 on non-windows platforms.
+  Used in templated usages of std::basic_view, etc.
+
+  Aliases for string types using Mso::char_traits_t
+
+  namespace Mso
+  {
+    char_traits_t
+    basic_string_view
+    basic_string
+  }
+
+  Example:
+
+  template<typename TChar>
+  bool TryHandleInput(const Mso::basic_string_view<TChar>& input) noexcept { ... }
+
+  template<typename TChar>
+  bool TryHandleInput(const Other::string_like_thing<TChar, Mso::char_traits_t<TChar>>& input) noexcept { ... }
+
+*/
+#pragma once
+#include <compilerAdapters/compilerWarnings.h>
+
+#ifdef MS_TARGET_ANDROID
+#include "wc16_wstring.h"
+#endif
+
+#include <string>
+
+namespace Mso {
+
+template <typename TChar>
+struct char_traits
+{
+  using type = std::char_traits<TChar>;
+};
+
+#if defined(MS_TARGET_ANDROID) || defined(MS_TARGET_APPLE)

```

Comment:
>defined(MS_TARGET_ANDROID) || defined(MS_TARGET_APPLE) [](http://example.com/codeflow?start=4&length=54)

It is better to use the MS_TARGET_POSIX instead: it combines Android, Mac, iOS, and Linux. See: [https://github.com/microsoft/mso/blob/main/tools/cmake/liblet.cmake](https://github.com/microsoft/mso/blob/main/tools/cmake/liblet.cmake)